roles/sft-server/meta/main.yml

```diff
@@ -16,4 +16,7 @@ galaxy_info:
   galaxy_tags:
     - SFT
 
-dependencies: []
+dependencies:
+  - role: cloudalchemy.node-exporter
+    vars:
+      node_exporter_web_listen_address: "127.0.0.1:{{ sft_node_exporter_port }}"
```

Comment:
Is there a reason we want more control over this?

> Also, you might want to pin the version of that dependency.

As I understand you cannot pin dependencies in this file. Let me know if there is a way.

Reply:
> Is there a reason we want more control over this?

IMHO having more control and explicit code would be a general aim :) (as long as it doesn't make things more complicated, but what does *complicated* even mean...)

> As I understand you cannot pin dependencies in this file. Let me know if there is a way.

You can indeed use the [`requirements.yml`](https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#dependencies) syntax.

Reply:
> IMHO having more control and explicit code would be desirable in general :) (as long as it doesn't make things more complicated, but what does complicated even mean...)

We can always take control of this when we need to anyways, i.e. if we think something needs to be changed with that role and it cannot be done here, we can just do what you're suggesting. But we don't need to do this now and it doesn't make it much different to do later. 

> You can indeed use the requirements.yml syntax.

Nice, I will give this a shot :+1:

Reply:
>> You can indeed use the requirements.yml syntax.
>
> Nice, I will give this a shot :+1:

This doesn't really do anything. Wherever this role will be called from, it will need a `requirements.yaml` to fetch node-exporter. I tried putting a different version in requirements.yaml and nothing complained about the mismatch. I vaguely remember talking to @jschaul about it and I think he was of the opinion that putting version there serves no purpose.

Reply:
:) It says:

> You can indeed use the requirements.yml __syntax__.

[Support for `requirements.yml` file in `./meta`](https://github.com/ansible/proposals/issues/57) only landed a couple of days ago.